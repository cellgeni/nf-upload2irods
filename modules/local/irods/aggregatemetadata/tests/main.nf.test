nextflow_process {
    name "Test Process: IRODS_AGGREGATEMETADATA"
    script "../main.nf"
    process "IRODS_AGGREGATEMETADATA"

    tag "modules"
    tag "irods"
    tag "irods/aggregatemetadata"

    setup {
        run("IRODS_GETMETADATA", alias: "CRAM") {
            script "../../getmetadata/main.nf"
            process {
                """
                input[0] = [ [id: "data_object_1"], "/seq/illumina/runs/41/41796/lane1/plex1/41796_1#1.cram" ]
                """
            }
        }

        run("IRODS_GETMETADATA", alias: "CELLRANGER_ARC_OUTPUT") {
            script "../../getmetadata/main.nf"
            process {
                """
                input[0] = [ [id: "data_object_1"], "/seq/illumina/cellranger-arc/cellranger-arc202_count_43c2d8dd1eaf98b635896165fd98ae3a" ]
                """
            }
        }

        run("IRODS_GETMETADATA", alias: "EMPTY") {
            script "../../getmetadata/main.nf"
            process {
                """
                input[0] = [ [id: "data_object_1"], "/archive/cellgeni/multiome" ]
                """
            }
        }
    }

    test("Cellranger ARC output metadata aggregation") {
        tag "cellranger_arc"

        when {
            process {
                """
                input[0] = CELLRANGER_ARC_OUTPUT.out.csv
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert process.out.versions },
                { assert process.out.csv },
                { assert process.out.json },
                { assert path(process.out.csv.get(0).get(1)).csv.rowCount == 1 },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test(".cram file metadata aggregation") {
        tag "cram"

        when {
            process {
                """
                input[0] = CRAM.out.csv
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert process.out.versions },
                { assert process.out.csv },
                { assert process.out.json },
                { assert path(process.out.csv.get(0).get(1)).csv.rowCount == 1 },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("Empty file output metadata aggregation") {
        tag "empty"

        when {
            process {
                """
                input[0] = EMPTY.out.csv
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert process.out.versions },
                { assert process.out.csv },
                { assert process.out.json },
                { assert path(process.out.csv.get(0).get(1)).size() == 0},
                { assert snapshot(process.out).match() }
            )
        }
    }
}