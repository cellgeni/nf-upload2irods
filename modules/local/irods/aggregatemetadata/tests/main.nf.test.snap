{
    "Empty file output metadata aggregation": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "data_object_1"
                        },
                        "metadata.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "data_object_1"
                        },
                        "metadata.json:md5,99914b932bd37a50b983c5e7c90ae93b"
                    ]
                ],
                "2": [
                    "versions.yml:md5,c7179ef11646eed9f4262bc31b48d134"
                ],
                "csv": [
                    [
                        {
                            "id": "data_object_1"
                        },
                        "metadata.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "json": [
                    [
                        {
                            "id": "data_object_1"
                        },
                        "metadata.json:md5,99914b932bd37a50b983c5e7c90ae93b"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,c7179ef11646eed9f4262bc31b48d134"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.4"
        },
        "timestamp": "2025-10-24T14:56:12.941011603"
    },
    ".cram file metadata aggregation": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "data_object_1"
                        },
                        "metadata.csv:md5,5ce876650fd0254c50d6c648c171a74c"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "data_object_1"
                        },
                        "metadata.json:md5,27c3fe81dd965b1ab87d800c9c41bfb4"
                    ]
                ],
                "2": [
                    "versions.yml:md5,c7179ef11646eed9f4262bc31b48d134"
                ],
                "csv": [
                    [
                        {
                            "id": "data_object_1"
                        },
                        "metadata.csv:md5,5ce876650fd0254c50d6c648c171a74c"
                    ]
                ],
                "json": [
                    [
                        {
                            "id": "data_object_1"
                        },
                        "metadata.json:md5,27c3fe81dd965b1ab87d800c9c41bfb4"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,c7179ef11646eed9f4262bc31b48d134"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.4"
        },
        "timestamp": "2025-10-24T14:55:40.705342216"
    },
    "Cellranger ARC output metadata aggregation": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "data_object_1"
                        },
                        "metadata.csv:md5,46ee296327f314c0cbb380418179fe3b"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "data_object_1"
                        },
                        "metadata.json:md5,85c49fdb2b6785753d79580d50d0db6c"
                    ]
                ],
                "2": [
                    "versions.yml:md5,c7179ef11646eed9f4262bc31b48d134"
                ],
                "csv": [
                    [
                        {
                            "id": "data_object_1"
                        },
                        "metadata.csv:md5,46ee296327f314c0cbb380418179fe3b"
                    ]
                ],
                "json": [
                    [
                        {
                            "id": "data_object_1"
                        },
                        "metadata.json:md5,85c49fdb2b6785753d79580d50d0db6c"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,c7179ef11646eed9f4262bc31b48d134"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.4"
        },
        "timestamp": "2025-10-24T14:43:00.769951028"
    }
}